README.md

```diff
@@ -1,2 +1,344 @@
 # optimizer
-AsyncAPI offers many different ways to reuse certain parts of the document like messages or schemas definitions or references to external files, not to even mention the traits. There is a need for a tool that can be plugged into any workflows and optimize documents that are generated from code, but not only.
+AsyncAPI offers many ways to reuse certain parts of the document like messages or schemas definitions or references to external files, not to even mention the traits. There is a need for a tool that can be plugged into any workflows and optimize documents that are generated from code, but not only.
```

Comment:
You have bug in the Readme.md because you show the unoptimized do - you have still duplicated:

```yaml
          - headers:
              type: object
              properties:
                my-app-header:
                  type: integer
                  minimum: 0
                  maximum: 100
```

You wrote in the PR description:

> only report generation for reuseComponents is implemented (report generation code for removeComponents and moveToComponents will be added in separate PRs).

so probably I don't understand something, or we truly have bug in Readme.md